### ISSUE TITLE
PromptEmbedding's random initialization leads to out-of-manifold embeddings that doesn't learn

### ISSUE THREAD

**Issue by macmacmacmac** [huggingface/peft#2816]
### Feature request

I already made a PR for this, but thought I should put this here anyways.

Currently, PromptEmbedding's random initialization uses `torch.nn.Embedding` to initialize its embeddings. However, because different models have different embedding spaces each with its own manifold of well-defined vocab embeddings, this naive initialization is highly unlikely to produce embeddings that land on the manifold, which leads to really poor learning empirically. From my testing naive random initialization reduces accuracy by almost a factor of 3.

To help visualize, here's a PCA of Llama 3.1 8b's vocab embeddings vs naively randomly initializing embeddings

<img width="702" height="701" alt="Image" src="https://github.com/user-attachments/assets/911865f4-ca40-44d9-983e-8eb19c512d37" />

VS. if we initialized embeddings by say randomly sampling tokens instead (suggested fix/feature). 

<img width="711" height="701" alt="Image" src="https://github.com/user-attachments/assets/2055c86f-e544-4cc3-8fe1-05521615c6a7" />

I implemented this as a new initialization option called RANDOM_DISCRETE. Hopefully the changes are backward compatible

### Your contribution

Here's my PR

https://github.com/huggingface/peft/pull/2815

**Comment by BenjaminBossan** [comment]
Resolved via #2815.
